pkgs/unified_analytics/lib/src/session.dart

```diff
@@ -3,28 +3,35 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:convert';
-import 'dart:io';
 
 import 'package:clock/clock.dart';
 import 'package:file/file.dart';
 import 'package:path/path.dart' as p;
 
+import 'analytics.dart';
 import 'constants.dart';
+import 'event.dart';
 import 'initializer.dart';
 
 class Session {
   final Directory homeDirectory;
   final FileSystem fs;
   final File sessionFile;
+  final Analytics _analyticsInstance;
 
   late int _sessionId;
   late int _lastPing;
 
+  /// Ensure we are only sending once per invocation for this class.
```

Comment:
Again, why only once?

Reply:
Same idea for the session handler here except we aren't dealing with the loop from the `map` method. With the session handler, if we have a problem with the `_refreshSessionData` method, then it will end up sending an  error event every time we call that method, which is every time we send an event.

So if a user had 100 events sent while an instance of `Analytics` is alive, then we will also have 100 error events